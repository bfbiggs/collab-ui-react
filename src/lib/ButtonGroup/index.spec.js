import React from 'react';
import { shallow, mount } from 'enzyme';
import { Button, ButtonGroup, Icon } from '@collab-ui/react';

describe('tests for <ButtonGroup />', () => {
  it('should match SnapShot', () => {
    const container = shallow(
      <ButtonGroup>
        <Button ariaLabel="test">1</Button>
      </ButtonGroup>);

    expect(container).toMatchSnapshot();
  });

  it('should focus button wrt focus prop', () => {
    const container = mount(
      <ButtonGroup focus={1}>
        <Button ariaLabel="test">1</Button>
        <Button ariaLabel="test" disabled>2</Button>
        <Button ariaLabel="test">3</Button>
      </ButtonGroup>);

    expect(container.find('button').at(2).props().tabIndex).toEqual(0);
  });

  it('ButtonGroup should have justified as true', () => {
    const container = mount(
      <ButtonGroup>
        <Button ariaLabel="test">1</Button>
        <Button ariaLabel="test">2</Button>
      </ButtonGroup>);

    expect(container.find('.cui-button-group').hasClass('cui-button-group--justified')).toEqual(true);
  });

  it('ButtonGroup type is set to space', () => {
    const container = mount(
      <ButtonGroup type="space">
        <Button ariaLabel="test">1</Button>
        <Button ariaLabel="test">2</Button>
      </ButtonGroup>);

    expect(container.find('.cui-button-group').hasClass('cui-button-group--space')).toEqual(true);
  });

  it('should not highlight the active button when highlightSelected is false', () => {
    const container = mount(
      <ButtonGroup highlightSelected={false}>
        <Button ariaLabel="test">1</Button>
        <Button ariaLabel="test">2</Button>
      </ButtonGroup>);

    container.find('button').at(0).simulate('click');
    expect(container.find('button').at(0).hasClass('active')).toEqual(false);
  });

  it('should apply an modifier to button when Button contains Icon as a children', () => {
    const container = mount(
      <ButtonGroup type="space">
        <Button ariaLabel="test">
          <Icon name="icon-arrow-left_12"/>
        </Button>
      </ButtonGroup>);

    expect(container.find('button').at(0).hasClass('cui-button--icon')).toEqual(true);
  });


  it('onClick should should mark the button as active', () => {
    const container = mount(
      <ButtonGroup focus={0}>
        <Button ariaLabel="test">1</Button>
        <Button ariaLabel="test">2</Button>
      </ButtonGroup>);
    container.find('button').at(1).simulate('click');
    expect(container.find('button').at(1).props().tabIndex).toEqual(0);
    expect(container.find('button').at(1).hasClass('active')).toEqual(true);
  });

  it('should handle keyBoard keys', () => {
    const container = mount(
      <ButtonGroup focus={0}>
        <Button ariaLabel="test">1</Button>
        <Button ariaLabel="test">2</Button>
      </ButtonGroup>);
    expect(container.find('button').at(0).props().tabIndex).toEqual(0);
    expect(container.find('button').at(0).hasClass('active')).toEqual(true);
    // right
    container.find('button').at(0).simulate('keydown', { keyCode: 39, which: 39 });
    expect(container.find('button').at(1).props().tabIndex).toEqual(0);
    // left
    container.find('button').at(1).simulate('keydown', { keyCode: 37, which: 37 });
    expect(container.find('button').at(0).props().tabIndex).toEqual(0);
    // up key
    container.find('button').at(0).simulate('keydown', { keyCode: 38, which: 38 });
    expect(container.find('button').at(1).props().tabIndex).toEqual(0);

    // enter key
    container.find('button').at(1).simulate('keydown', { keyCode: 13, which: 13 });
    expect(container.find('button').at(1).hasClass('active')).toEqual(true);
  });

  it('should throw error if child is not an Button', () => {
    try {
      shallow(
          <ButtonGroup>
            <div/>
          </ButtonGroup>
      );
    } catch(e) {
      expect(e.message).toEqual('ButtonGroup should only contain Buttons as children.');
    }
  });

});
