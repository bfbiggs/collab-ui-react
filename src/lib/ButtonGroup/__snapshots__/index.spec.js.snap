// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for <ButtonGroup /> should match SnapShot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonGroup
    activeIndex={null}
    ariaLabel=""
    className=""
    focusOnLoad={false}
    highlightSelected={true}
    justified={true}
    onSelect={null}
    type=""
  >
    <Button
      active={false}
      ariaLabel="test"
      ariaLabelledBy=""
      circle={false}
      className=""
      color=""
      containerLarge={false}
      disabled={false}
      expand={false}
      href=""
      index={null}
      label=""
      large={false}
      loading={false}
      onClick={null}
      size={36}
      style={Object {}}
      tag="button"
      type="button"
    >
      1
    </Button>
  </ButtonGroup>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "aria-label": "",
      "children": Array [
        <Button
          active={false}
          ariaLabel="test"
          ariaLabelledBy=""
          circle={false}
          className=""
          color=""
          containerLarge={false}
          disabled={false}
          expand={false}
          href=""
          index={0}
          label=""
          large={false}
          loading={false}
          onClick={null}
          size={36}
          style={Object {}}
          tag="button"
          type="button"
        >
          1
        </Button>,
      ],
      "className": "cui-button-group cui-button-group--justified",
      "role": "group",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": ".0",
        "nodeType": "class",
        "props": Object {
          "active": false,
          "ariaLabel": "test",
          "ariaLabelledBy": "",
          "children": "1",
          "circle": false,
          "className": "",
          "color": "",
          "containerLarge": false,
          "disabled": false,
          "expand": false,
          "href": "",
          "index": 0,
          "label": "",
          "large": false,
          "loading": false,
          "onClick": null,
          "size": 36,
          "style": Object {},
          "tag": "button",
          "type": "button",
        },
        "ref": null,
        "rendered": "1",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "aria-label": "",
        "children": Array [
          <Button
            active={false}
            ariaLabel="test"
            ariaLabelledBy=""
            circle={false}
            className=""
            color=""
            containerLarge={false}
            disabled={false}
            expand={false}
            href=""
            index={0}
            label=""
            large={false}
            loading={false}
            onClick={null}
            size={36}
            style={Object {}}
            tag="button"
            type="button"
          >
            1
          </Button>,
        ],
        "className": "cui-button-group cui-button-group--justified",
        "role": "group",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": ".0",
          "nodeType": "class",
          "props": Object {
            "active": false,
            "ariaLabel": "test",
            "ariaLabelledBy": "",
            "children": "1",
            "circle": false,
            "className": "",
            "color": "",
            "containerLarge": false,
            "disabled": false,
            "expand": false,
            "href": "",
            "index": 0,
            "label": "",
            "large": false,
            "loading": false,
            "onClick": null,
            "size": 36,
            "style": Object {},
            "tag": "button",
            "type": "button",
          },
          "ref": null,
          "rendered": "1",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
